src/libostree/ostree-kernel-args.c

```diff
@@ -163,6 +163,49 @@ strcmp0_equal (gconstpointer v1, gconstpointer v2)
   return g_strcmp0 (v1, v2) == 0;
 }
 
+/* Split string with whitespace, but keep quotes.
+   For example, "test=\"1 2\"" will be saved as whole.
+ */
+static char **
+split_whitespace (const char *str)
+{
+  gboolean quoted = FALSE;
+  g_return_val_if_fail (str != NULL, NULL);
+
+  GPtrArray *strv = g_ptr_array_new ();
+
+  guint len = strlen (str);
```

Comment:
Technically this should be `size_t` not `guint`; they're different sizes.  A giant pitfall in C is that it will happily silently cast numeric types.  Of course, we shouldn't be dealing with >4G strings but still.